drivers/i2c/i2c_npcx_controller.c

```diff
@@ -1105,16 +1150,29 @@ int npcx_i2c_ctrl_target_register(const struct device *i2c_dev,
 
 	/* Reconfigure SMBCTL1 */
 	inst->SMBCTL1 |= BIT(NPCX_SMBCTL1_NMINTE) | BIT(NPCX_SMBCTL1_INTEN);
+
+	/* Enable irq of smb wake-up event */
+	if (IS_ENABLED(CONFIG_PM)) {
+
+		/* Enable SMB wake up detection */
+		npcx_i2c_target_start_wk_enable(idx_ctrl, true);
+		/* Enable start detect in IDLE */
+		inst->SMBCTL3 |= BIT(NPCX_SMBCTL3_IDL_START);
+		/* Enable SMB's MIWU interrupts */
+		npcx_miwu_irq_enable(&config->smb_wui);
+	}
 	i2c_ctrl_irq_enable(i2c_dev, 1);
 
 	return 0;
 }
 
 int npcx_i2c_ctrl_target_unregister(const struct device *i2c_dev,
-				   struct i2c_target_config *target_cfg)
+				   struct i2c_target_config *target_cfg, uint8_t port)
 {
 	struct smb_reg *const inst = HAL_I2C_INSTANCE(i2c_dev);
+	const struct i2c_ctrl_config *const config = i2c_dev->config;
 	struct i2c_ctrl_data *const data = i2c_dev->data;
+	int idx_ctrl = (port & 0xF0) >> 4;
```

Comment:
Non-blocking comment - the `port` property is overloaded to contain the controller and the port index, but this isn't documented in the nuvoton,npcx-i2c.yaml file.

Please update the yaml file in a follow up PR.  It would also be cleaner if you created a helper macro in the dt-bindings.
```
#define NPCX_I2C_CTRL_PORT(ctrl, port) ((ctrl & 0xf << 4) | port & 0xf)

...
i2c1_0: io_i2c_ctrl1_port0 {
    port = <NPCX_I2C_CTRL_PORT(1,0)>;
};
```

Reply:
Will update in the next PR, thanks.